src/sentry/taskworker/pending_tasks.py

```diff
@@ -0,0 +1,23 @@
+from dataclasses import dataclass
+from enum import Enum
+
+from sentry_protos.hackweek_team_no_celery_pls.v1alpha.pending_task_pb2 import (
+    PendingTask as PendingTaskProto,
+)
+
+
+class PendingTaskState(Enum):
+    PENDING = "pending"
+    PROCESSING = "processing"
+    COMPLETE = "complete"
+    FAILURE = "failure"
+    RETRY = "retry"
+
+
+@dataclass
+class PendingTask:
+    proto_task: PendingTaskProto
```

Comment:
:+1: Decorating with a 'domain object' is a good path to take. Copying all the values out of the protobuf will be expensive in aggregate.

We might want to add methods/properties to this object that proxy to the `proto_task` and have the `proto_task` be a private property.